### PR TITLE
Support uuid in storage service

### PR DESCRIPTION
**Pull request by critical27**
`UUID` key format: type(1) + partId(3) + name.
The value is the corresponding vertex id.

Waiting on #914.
Subtask of #901.

### CODE REVIEW

File: src/storage/GetUUIDProcessor.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        auto partId = req.get_part_id();
+        auto name = req.get_name();
+        auto key = NebulaKeyUtils::uuidKey(partId, name.c_str());
+        std::string val;
+        VertexID vId;
+        auto ret = kvstore_->get(spaceId, partId, key, &val);
+        // try to get the corresponding vertex id
+        if (ret != kvstore::ResultCode::SUCCEEDED) {
+            // need to generate new vertex id of this uuid
+            std::hash<std::string> hashFunc;
```
Comment (dangleptr) [inline src/storage/GetUUIDProcessor.h]: We'd better use different hash function with partition when generating UUID.

File: src/storage/GetUUIDProcessor.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            auto hashValue = hashFunc(name);
+            auto now = time::WallClock::fastNowInMicroSec();
+            vId = (hashValue & hashMask) | (now & timeMask);
+            val.append(reinterpret_cast<char*>(&vId), sizeof(VertexID));
+            std::vector<kvstore::KV> data;
+            data.emplace_back(key, val);
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/storage/GetUUIDProcessor.h]: Good catch.

File: src/storage/GetUUIDProcessor.h (hunk 1 above)
Comment (critical27) [reply src/storage/GetUUIDProcessor.h]: 👍

File: src/storage/GetUUIDProcessor.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        auto key = NebulaKeyUtils::uuidKey(partId, name.c_str());
+        std::string val;
+        VertexID vId;
+        auto ret = kvstore_->get(spaceId, partId, key, &val);
+        // try to get the corresponding vertex id
+        if (ret != kvstore::ResultCode::SUCCEEDED) {
+            // need to generate new vertex id of this uuid
+            MurmurHash2 hashFunc;
+            auto hashValue = hashFunc(name);
+            auto now = time::WallClock::fastNowInMicroSec();
```
Comment (dangleptr) [inline src/storage/GetUUIDProcessor.h]: for int32 we’d better use seconds

File: src/common/base/NebulaKeyUtils.cpp [hunk 4]
```diff
@@ -68,6 +68,16 @@ std::string NebulaKeyUtils::systemPartKey(PartitionID partId) {
     return key;
 }
 
+// static
+std::string NebulaKeyUtils::uuidKey(PartitionID partId, const folly::StringPiece& name) {
+    std::string key;
+    key.reserve(sizeof(PartitionID) + name.size());
+    int32_t item = (partId << 8) | (NebulaKeyType::kUUID);
+    key.append(reinterpret_cast<const char*>(&item), sizeof(int32_t))
+       .append(name.str());
```
Comment (dangleptr) [inline src/common/base/NebulaKeyUtils.cpp]: Extra copy here

File: src/storage/GetUUIDProcessor.h (hunk 3 above)
Comment (critical27) [reply src/storage/GetUUIDProcessor.h]: If we use second, I suppose we would have high conflict rate? Cuz If we generate many uuid in a partition at the exact same second, the time will be all the same.

File: src/common/base/NebulaKeyUtils.cpp (hunk 4 above)
Comment (critical27) [reply src/common/base/NebulaKeyUtils.cpp]: Good catch

File: src/storage/GetUUIDProcessor.h (hunk 3 above)
Comment (dangleptr) [reply src/storage/GetUUIDProcessor.h]: we can’t use int32 to represent micro seconds, it has more conflicts

Comment (critical27) [reply src/storage/GetUUIDProcessor.h]: Ok

**Pull request merged**